hibernate-core/src/main/java/org/hibernate/sql/results/internal/RowTransformerConstructorImpl.java

```diff
@@ -33,8 +35,12 @@ public RowTransformerConstructorImpl(Class<T> type, TupleMetadata tupleMetadata)
 		for (int i = 0; i < elements.size(); i++) {
 			sig[i] = resolveElementJavaType( elements.get( i ) );
 		}
+		if ( sig.length == 1 && sig[0] == null ) {
+			// Can not (properly) resolve constructor for single null element
+			throw new InstantiationException( "Cannot instantiate query result type ", type );
```

Comment:
Please add a more descriptive error message for this case. Something like `Cannot instantiate query result type, because argument types are unknown`

Reply:
Done